charts/stable/nextcloud/templates/_hpb.tpl

```diff
@@ -108,6 +108,15 @@ command:
     {{ if .Values.imaginary.preview_font }}php /var/www/html/occ config:system:set enabledPreviewProviders {{ $c }} --value='OC\Preview\Font'{{ $c = add1 $c }}{{ end }}
     {{- end }}
 
+    # Set overwrite.cli.uri
+    {{- $url := (  printf "http://%s/" ( .Values.env.AccessIP | default ( printf "%v-%v" .Release.Name "nextcloud" ) ) ) }}
+    {{- if .Values.ingress.main.enabled }}
+      {{- with (first .Values.ingress.main.hosts) }}
+      {{- $url = (  printf "https://%s/" .host ) }}
+      {{- end }}
+    {{- end }}
```

Comment:
this also needs to take into account scenario's where the main service main port, is not 80.